File: FragmentPractice/app/src/main/java/com/example/fragmentpractice/MainActivity.java
package com.example.fragmentpractice;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
/**通过限定符，mainActivity根据屏幕宽度加载不同的布局文件activity_main.xml；
 * 不同activity均已静态加载显示新闻标题列表的NewsTitleFragment，其通过recyclerView显示列表，
 * 该碎片自行判断单双页模式，然后通过不同方式跳转到新闻详情页：
 * 一、双页模式，直接将新闻数据传给显示新闻详情页的NewsContentFragment，同页展示；
 * 二、单页模式，启动新的activity，然后在该activity中加载显示新闻详情页的NewsContentFragment，实现换页展示。
 * */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: AllInOne/app/src/main/java/com/justap/floc/MainActivity.kt
package com.justap.floc

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.GridLayoutManager
import com.justap.floc.brocastRcver.BroadcastMainAct
import com.justap.floc.contentProvider.ContentProviderActivity
import com.justap.floc.persistenceStorage.MainPersistenceActivity
import com.justap.floc.serviceAndThread.ThreadTestActivity
import floc.R
import kotlinx.android.synthetic.main.activity_main.*

//《第一行代码》实例练习

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val entryList = mutableListOf<EntryItem>()
        initList(entryList)
        val layoutManager = GridLayoutManager(this,3)
        rv_menu.layoutManager = layoutManager
        rv_menu.adapter = MenuAdapter(entryList)
}

    private fun initList(entryList: MutableList<EntryItem>) {

        entryList.add(EntryItem("Service&Thread",R.drawable.star_empty_black){
            startActivity(Intent(this,ThreadTestActivity::class.java))
        })

        entryList.add(EntryItem("ContentProvider",R.drawable.star_empty_black){
            startActivity(Intent(this,ContentProviderActivity::class.java))
        })

        entryList.add(EntryItem("持久化技术", R.drawable.star_empty_grey) {
            startActivity(Intent(this, MainPersistenceActivity::class.java))
        })

        entryList.add(EntryItem("广播\nBroadcast", R.drawable.star_empty_grey) {
            startActivity(Intent(this, BroadcastMainAct::class.java))
        })

        for(i in 1..30){
            entryList.add(EntryItem("测试$i", R.drawable.star_empty_grey) {
                Toast.makeText(this,"测试$i",Toast.LENGTH_SHORT).show()
            })
        }
    }

}